view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java

```diff
@@ -368,28 +374,106 @@ public static boolean sendPinotTableCreationRequest(
     return sendPinotTableCreationRequest(
         pinotTableSpec.getControllerHost(),
         pinotTableSpec.getControllerPort(),
-        tableConfig.toJsonString());
+        tableConfig.toJsonString(),
+        tableConfig.getTableName());
   }
 
   public static boolean sendPinotTableCreationRequest(
-      String controllerHost, String controllerPort, final String tableConfig) {
+      String controllerHost,
+      String controllerPort,
+      final String tableConfig,
+      final String tableName) {
+    String controllerAddress = getControllerAddressForTableCreate(controllerHost, controllerPort);
+    LOGGER.info("Trying to send table creation request {} to {}. ", tableConfig, controllerAddress);
+    int responseCode = sendRequest("POST", controllerAddress, tableConfig);
+    if (responseCode == 409) {
+      LOGGER.info("Trying to update table with request {} to {}. ", tableConfig, controllerAddress);
+      sendRequest(
+          "PUT",
+          getControllerAddressForTableUpdate(controllerHost, controllerPort, tableName),
+          tableConfig);
+    }
+    return true;
+  }
+
+  /**
+   * @param requestMethod: Type of request (PUT/GET/POST/DELTE)
+   * @param urlString: Api endpoint
+   * @param payload: Payload
+   * @return Status Code
+   */
+  public static int sendRequest(String requestMethod, String urlString, String payload) {
```

Comment:
Do we have to impl this ourselves? Nothing in the libraries to help?

Reply:
Actually previously, we were using Pinot library for creating/updating library. But there implementation is a bit buggy in the sense that we can't update OFFLINE tables (though REALTIME can be) via POST request to api. So we are explicitly initiating the request.

Reply:
ok